From dfec756b44260b00ce02e924cb8cbfeaa5332170 Mon Sep 17 00:00:00 2001
From: Motomu Utsumi <motomuman@google.com>
Date: Thu, 16 Feb 2023 17:00:01 +0900
Subject: [PATCH] cronet: Rename QuicOptions to QuicParams

Fix UserHandleName API linter error for QuicOptions
```
QuicOptions.java:22: error: Classes holding a set of parameters should be
called `FooParams`, was `QuicOptions` [UserHandleName]
```

Test: TH
Bug: 265674359
Change-Id: I67e7300c5a96e0a18563186ebee79ed2ecd35e1d
---
 components/cronet/android/BUILD.gn             |  2 +-
 .../net/http/ExperimentalHttpEngine.java       |  4 ++--
 .../api/src/android/net/http/HttpEngine.java   | 18 +++++++++---------
 .../android/net/http/IHttpEngineBuilder.java   |  2 +-
 .../http/{QuicOptions.java => QuicParams.java} | 14 +++++++-------
 5 files changed, 20 insertions(+), 20 deletions(-)
 rename components/cronet/android/api/src/android/net/http/{QuicOptions.java => QuicParams.java} (98%)

diff --git a/components/cronet/android/BUILD.gn b/components/cronet/android/BUILD.gn
index db1ee1bb..5263b996 100644
--- a/components/cronet/android/BUILD.gn
+++ b/components/cronet/android/BUILD.gn
@@ -241,7 +241,7 @@ android_library("cronet_api_java") {
     "api/src/android/net/http/NetworkQualityRttListener.java",
     "api/src/android/net/http/NetworkQualityThroughputListener.java",
     "api/src/android/net/http/QuicException.java",
-    "api/src/android/net/http/QuicOptions.java",
+    "api/src/android/net/http/QuicParams.java",
     "api/src/android/net/http/RequestFinishedInfo.java",
     "api/src/android/net/http/UploadDataProvider.java",
     "api/src/android/net/http/UploadDataSink.java",
diff --git a/components/cronet/android/api/src/android/net/http/ExperimentalHttpEngine.java b/components/cronet/android/api/src/android/net/http/ExperimentalHttpEngine.java
index 04d93e8e..8a195f21 100644
--- a/components/cronet/android/api/src/android/net/http/ExperimentalHttpEngine.java
+++ b/components/cronet/android/api/src/android/net/http/ExperimentalHttpEngine.java
@@ -184,8 +184,8 @@ public abstract class ExperimentalHttpEngine extends HttpEngine {
         }
 
         @Override
-        @QuicOptions.Experimental
-        public Builder setQuicOptions(QuicOptions options) {
+        @QuicParams.Experimental
+        public Builder setQuicParams(QuicParams options) {
             // If the delegate builder supports enabling connection migration directly, just use it
             if (mBuilderDelegate.getSupportedConfigOptions().contains(
                     IHttpEngineBuilder.QUIC_OPTIONS)) {
diff --git a/components/cronet/android/api/src/android/net/http/HttpEngine.java b/components/cronet/android/api/src/android/net/http/HttpEngine.java
index 04c399cb..27a0da94 100644
--- a/components/cronet/android/api/src/android/net/http/HttpEngine.java
+++ b/components/cronet/android/api/src/android/net/http/HttpEngine.java
@@ -286,27 +286,27 @@ public abstract class HttpEngine {
 
         /**
          * Configures the behavior of the HTTP stack when using QUIC. For more details, see
-         * documentation of {@link QuicOptions} and the individual methods
-         * of {@link QuicOptions.Builder}.
+         * documentation of {@link QuicParams} and the individual methods
+         * of {@link QuicParams.Builder}.
          *
          * <p>Only relevant if {@link #setEnableQuic(boolean)} is enabled.
          *
          * @return the builder to facilitate chaining.
          */
-        @QuicOptions.Experimental
-        public Builder setQuicOptions(QuicOptions quicOptions) {
-            mBuilderDelegate.setQuicOptions(quicOptions);
+        @QuicParams.Experimental
+        public Builder setQuicParams(QuicParams quicParams) {
+            mBuilderDelegate.setQuicOptions(quicParams);
             return this;
         }
 
         /**
-         * @see #setQuicOptions(QuicOptions)
+         * @see #setQuicParams(QuicParams)
          *
          * {@hide}
          */
-        @QuicOptions.Experimental
-        public Builder setQuicOptions(QuicOptions.Builder quicOptionsBuilder) {
-            return setQuicOptions(quicOptionsBuilder.build());
+        @QuicParams.Experimental
+        public Builder setQuicParams(QuicParams.Builder quicParamsBuilder) {
+            return setQuicParams(quicParamsBuilder.build());
         }
 
         /**
diff --git a/components/cronet/android/api/src/android/net/http/IHttpEngineBuilder.java b/components/cronet/android/api/src/android/net/http/IHttpEngineBuilder.java
index ca1e94e6..1559489f 100644
--- a/components/cronet/android/api/src/android/net/http/IHttpEngineBuilder.java
+++ b/components/cronet/android/api/src/android/net/http/IHttpEngineBuilder.java
@@ -50,7 +50,7 @@ public abstract class IHttpEngineBuilder {
         return this;
     }
 
-    public IHttpEngineBuilder setQuicOptions(QuicOptions quicOptions) {
+    public IHttpEngineBuilder setQuicOptions(QuicParams quicParams) {
         return this;
     }
 
diff --git a/components/cronet/android/api/src/android/net/http/QuicOptions.java b/components/cronet/android/api/src/android/net/http/QuicParams.java
similarity index 98%
rename from components/cronet/android/api/src/android/net/http/QuicOptions.java
rename to components/cronet/android/api/src/android/net/http/QuicParams.java
index 11899ed5..e1c3e55f 100644
--- a/components/cronet/android/api/src/android/net/http/QuicOptions.java
+++ b/components/cronet/android/api/src/android/net/http/QuicParams.java
@@ -19,7 +19,7 @@ import java.util.Set;
  * {@link HttpEngine.Builder#setEnableQuic(boolean)} to enable / disable QUIC for
  * the HTTP engine.
  */
-public class QuicOptions {
+public class QuicParams {
     private final Set<String> mQuicHostAllowlist;
     private final Set<String> mEnabledQuicVersions;
 
@@ -58,7 +58,7 @@ public class QuicOptions {
 
     private final Set<String> mExtraQuicheFlags;
 
-    QuicOptions(Builder builder) {
+    QuicParams(Builder builder) {
         this.mQuicHostAllowlist =
                 Collections.unmodifiableSet(new LinkedHashSet<>(builder.mQuicHostAllowlist));
         this.mEnabledQuicVersions =
@@ -271,7 +271,7 @@ public class QuicOptions {
     }
 
     /**
-     * Create a new {@code QuicOptions} builder.
+     * Create a new {@code QuicParams} builder.
      *
      * {@hide}
      */
@@ -280,7 +280,7 @@ public class QuicOptions {
     }
 
     /**
-     * Builder for {@link QuicOptions}.
+     * Builder for {@link QuicParams}.
      */
     public static final class Builder {
         private final Set<String> mQuicHostAllowlist = new LinkedHashSet<>();
@@ -624,12 +624,12 @@ public class QuicOptions {
         }
 
         /**
-         * Creates and returns the final {@link QuicOptions} instance, based on the values
+         * Creates and returns the final {@link QuicParams} instance, based on the values
          * in this builder.
          */
         @NonNull
-        public QuicOptions build() {
-            return new QuicOptions(this);
+        public QuicParams build() {
+            return new QuicParams(this);
         }
     }
 
-- 
2.39.2.637.g21b0678d19-goog

